src/konserve/encryptor.cljc

```diff
@@ -12,38 +12,49 @@
   (-serialize [_ bytes write-handlers val]
     (-serialize serializer bytes write-handlers val)))
 
-(defn null-encryptor [_store-key _config]
+(defn null-encryptor [_config]
   (fn [serializer]
     (NullEncryptor. serializer)))
 
 ;; Following advise at
 ;; https://crypto.stackexchange.com/questions/84439/is-it-dangerous-to-encrypt-lots-of-small-files-with-the-same-key
 
-(defn store-key->iv [store-key]
-  (subvec (vec (edn-hash store-key)) 0 16))
+(defn get-iv [salt key]
+  (subvec (vec (edn-hash ["initial-value" salt key])) 0 16))
 
-(defrecord AESEncryptor [serializer store-key key]
+(defn get-key [salt key]
+  ["key" salt key])
+
+;; TODO cljs support incomplete, prepending of salt is missing
+(defrecord AESEncryptor [serializer key]
   PStoreSerializer
   (-deserialize [_ read-handlers bytes]
-    (let [decrypted (decrypt [store-key key] #?(:clj (.readAllBytes ^ByteArrayInputStream bytes) :cljs bytes)
-                             :iv (store-key->iv store-key))]
+    (let [salt-array (byte-array 64)
+          _ (.read ^ByteArrayInputStream bytes salt-array)
+          salt (map int salt-array)
+          decrypted (decrypt (get-key salt key)
+                             #?(:clj (.readAllBytes ^ByteArrayInputStream bytes) :cljs bytes)
+                             :iv (get-iv salt key))]
       (-deserialize serializer read-handlers (ByteArrayInputStream. decrypted))))
```

Comment:
I'd say the constructor for `ByteArrayInputStream` will only work in clojure, right?

Reply:
Yes, I reached out to @pkpkpk, maybe he can help me write idiomatic ClojureScript/JavaScript here, if not then I will have to fix ClojureScript later. It is mostly about concatenating buffers.